### PR TITLE
feat: secrets encryption

### PR DESCRIPTION
**Pull request by MauroCicerchia**
This pull request introduces encryption for environment variable storage, ensuring that sensitive data is securely stored in the database. It adds a new AES-256-GCM encryption component, integrates it with the environment storage logic, updates the API and types accordingly, and provides comprehensive unit tests for the encryption functionality.

**Encryption Infrastructure**

* Added a new `IEncryptionComponent` with AES-256-GCM encryption and decryption methods, including error handling via a `DecryptionError` class (`src/adapters/encryption/component.ts`, `src/adapters/encryption/types.ts`, `src/adapters/encryption/index.ts`).
* Introduced `ENCRYPTION_KEY` configuration in `.env.default` for the encryption component to use.

**Integration with Environment Storage**

* Updated `env-storage` to encrypt values before storing and decrypt them when retrieving, replacing previous plaintext logic. The `setValue` method now returns `void` instead of the stored value (`src/adapters/env-storage/component.ts`, `src/adapters/env-storage/types.ts`).

**API and Test Updates**

* Changed the upsert environment variable API to return HTTP 204 (No Content) instead of 200 (`src/controllers/handlers/env-storage/upsert-env-storage.ts`).
* Updated integration tests to expect the new 204 response and verify correct value retrieval (`test/integration/env-storage/upsert-env-storage-controller.spec.ts`).

**Testing and Validation**

* Added a comprehensive unit test suite for the encryption component, covering correct encryption/decryption, error handling, and edge cases (`test/unit/adapters/encryption.spec.ts`).

### CODE REVIEW

File: test/unit/adapters/encryption.spec.ts [hunk 1]
```diff
@@ -0,0 +1,239 @@
+import { randomBytes } from 'crypto'
+import { createEncryptionComponent } from '../../../src/adapters/encryption'
+import { DecryptionError } from '../../../src/adapters/encryption/types'
+import type { IEncryptionComponent } from '../../../src/adapters/encryption'
+import type { AppComponents } from '../../../src/types'
+
+describe('createEncryptionComponent', () => {
```
Comment (kevinszuchet) [inline test/unit/adapters/encryption.spec.ts]:
```suggestion
describe('Encryption Component', () => {
```
